### PR TITLE
[#1148] changed instanceIsolation filter oder before zoneAware filter

### PR DESCRIPTION
**Pull request by chengyouling**

### CODE REVIEW

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java [hunk 1]
```diff
@@ -88,6 +91,6 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
 
   @Override
   public int getOrder() {
-    return Ordered.LOWEST_PRECEDENCE;
+    return env.getProperty("spring.cloud.servicecomb.instance.isolation.filter.order", Integer.class, -3);
```
Comment (liubao68) [inline spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java:94]: 1. `spring.cloud.loadbalance.filter.order.instance-isolation`
2. int.class

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/AffinityTagServiceInstanceFilter.java [hunk 2]
```diff
@@ -54,6 +59,6 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
 
   @Override
   public int getOrder() {
-    return AFFINITY_TAG_ORDER;
+    return env.getProperty("spring.cloud.servicecomb.affinity.tag.filter.order", Integer.class, 0);
```
Comment (liubao68) [inline spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/AffinityTagServiceInstanceFilter.java:62]: 1. `spring.cloud.loadbalance.filter.order.affinity-tag`
2. int.class

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/ZoneAwareServiceInstanceFilter.java [hunk 3]
```diff
@@ -51,7 +56,7 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
 
   @Override
   public int getOrder() {
-    return ZONE_AWARE_ORDER;
+    return env.getProperty("spring.cloud.servicecomb.zone.aware.filter.order", Integer.class, -2);
```
Comment (liubao68) [inline spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/ZoneAwareServiceInstanceFilter.java:59]: 1. `spring.cloud.loadbalance.filter.zone-aware.order`
2. int.class

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webflux/WebFluxServiceInstanceFilter.java [hunk 4]
```diff
@@ -65,6 +69,6 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
 
   @Override
   public int getOrder() {
-    return CANARY_ORDER;
+    return env.getProperty("spring.cloud.servicecomb.web.router.filter.order", Integer.class, -1);
```
Comment (liubao68) [inline spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webflux/WebFluxServiceInstanceFilter.java:72]: 1. `spring.cloud.loadbalance.filter.router.order`
2. int.class

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java [hunk 5]
```diff
@@ -51,6 +55,6 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
 
   @Override
   public int getOrder() {
-    return CANARY_ORDER;
+    return env.getProperty("spring.cloud.servicecomb.web.router.filter.order", Integer.class, -1);
```
Comment (liubao68) [inline spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java:58]: 1. `spring.cloud.loadbalance.filter.router.order`
2. int.class

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java (hunk 1 above)
Comment (chengyouling) [reply spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java:94]: fixed

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/AffinityTagServiceInstanceFilter.java (hunk 2 above)
Comment (chengyouling) [reply spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/AffinityTagServiceInstanceFilter.java]: fixed

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/ZoneAwareServiceInstanceFilter.java (hunk 3 above)
Comment (chengyouling) [reply spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/ZoneAwareServiceInstanceFilter.java]: fixed

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webflux/WebFluxServiceInstanceFilter.java (hunk 4 above)
Comment (chengyouling) [reply spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webflux/WebFluxServiceInstanceFilter.java]: fixed

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java (hunk 5 above)
Comment (chengyouling) [reply spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java]: fixed

File: spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java [hunk 6]
```diff
@@ -51,6 +55,6 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
 
   @Override
   public int getOrder() {
-    return CANARY_ORDER;
+    return env.getProperty("spring.cloud.loadbalance.filter.router.order", int.class, -1);
```
Comment (liubao68) [inline spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java:58]: Use order 100, 200, or -100, -200 is better.

Comment (chengyouling) [reply spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/loadbalancer/webmvc/WebMvcServiceInstanceFilter.java]: fixed

**Review by liubao68 — APPROVED** [review]

**Pull request merged**